### PR TITLE
Redesign: Filter button form component

### LINKED ISSUE #6463: Redesign: Update background filter panel
## Feature Description
See [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3404%3A198113)

- [x] Update title
- [x] Update color input
- [x] Update the name of the panel

Note: Keep the same filter toggle buttons for now as in the old panel. Updating those happen in #6527.

## Additional Context

Screenshot 2021-02-22 at 12 16 02

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### LINKED ISSUE #6499: Redesign: Generic Filter Button component
## Feature Description

See [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3360%3A63235) 

- [ ] Create new component in design system
- [ ] Accept optional image URL
- [ ] Add to storybook

## Additional Context
Screen Shot 2021-02-24 at 21 02 36

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### LINKED ISSUE #6527: Redesign: Update image filter buttons to proper design system component
## Feature Description

Once #6499 is implemented, update the filter panel to use this component.

See [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3404%3A198113)

## Additional Context

Screenshot 2021-02-22 at 12 16 02
---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by miina**
## Context
Adds filter button to the editors form components.

## Relevant Technical Choices

Did not add this to the design-panel since it seems to be very editor-specific and use functionality from the editor (e.g. patterns)

## User-facing changes
| Before | After |
|--------|------|
| Screenshot 2021-03-25 at 16 12 27 | Screenshot 2021-03-25 at 16 09 33 |
| Screenshot 2021-03-25 at 16 12 19 | Screenshot 2021-03-25 at 16 09 21 |

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

See [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3389%3A64503)

1. Add background media (preferably light, to notice the difference better)
2. Verify the Filters panel looks as in Figma but using the actual image of the selected element.
3. Verify the selected filter displays a light outline
4. Verify that when tabbing through the filters the focus line is displayed per Figma.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.
## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6499 
Fixes #6527

### CODE REVIEW

File: assets/src/edit-story/components/panels/design/filter/filter.js [hunk 1]
```diff
@@ -47,28 +48,30 @@ function FilterPanel({ selectedElements, pushUpdate }) {
...
-              iconHeight={16}
+              filter={convertOverlay(overlay, OverlayType.NONE, type)}
               aria-label={sprintf(
                 /* translators: %s: Filter type */
                 __('Set filter: %s', 'web-stories'),
                 label
               )}
-            />
+            >
+              <LayerIcon element={selectedElements[0]} />
```
Comment (barklund) [inline assets/src/edit-story/components/panels/design/filter/filter.js:73]: Oh sneaky, didn't consider re-using this component. Seems to work great!

File: assets/src/edit-story/components/panels/design/filter/karma/filter.karma.js [hunk 2]
```diff
@@ -108,32 +108,32 @@ describe('Filter Panel', () => {
...
     it('should render correct overlay when clicking "linear"', async () => {
-      await fixture.events.click(filterPanel.linear.button);
+      await fixture.events.click(filterPanel.linear);
 
       const overlay = await waitFor(getBackgroundElementOverlay);
       expect(overlay).toBeTruthy();
       expect(overlay).toHaveStyle(
         'background-image',
-        'linear-gradient(rgba(0, 0, 0, 0) 40%, rgba(0, 0, 0, 0.9) 100%)'
+        'linear-gradient(rgba(0, 0, 0, 0) 0%, rgba(0, 0, 0, 0.63) 100%)'
```
Comment (barklund) [inline assets/src/edit-story/components/panels/design/filter/karma/filter.karma.js]: New defaults?

File: assets/src/edit-story/karma/fixture/containers/designPanel/filter.js [hunk 3]
```diff
@@ -30,34 +29,18 @@ export class Filter extends AbstractPanel {
   }
 
   get none() {
-    return this._get(
-      this.getByRole('checkbox', { name: /Set filter: None/i }),
-      'none',
-      Toggle
-    );
+    return this.getByRole('button', { name: /Set filter: None/i });
```
Comment (barklund) [inline assets/src/edit-story/karma/fixture/containers/designPanel/filter.js]: I actually think we should update these labels now that we're at it. Probably to something like: "No filter", "Tint filter", "Linear filter", "Vignette filter" or something like that. The verb-version here is misunderstood ARIA.

**Review by barklund — APPROVED** [review]
Looks great - just a couple of comments.

File: assets/src/edit-story/components/panels/design/filter/karma/filter.karma.js (hunk 2 above)
Comment (miina) [reply assets/src/edit-story/components/panels/design/filter/karma/filter.karma.js]: Yep, as far as I could see but I'll double-check the values.

Comment (miina) [reply assets/src/edit-story/components/panels/design/filter/karma/filter.karma.js]: Ok, looks like I had incorrect values here as a result due to alpha and the `size` -- updated these too, should match Figma now.

File: assets/src/edit-story/components/form/filterToggle/stories/index.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
```
Comment (swissspidy) [inline assets/src/edit-story/components/form/filterToggle/stories/index.js]: Unneeded `React` import

Comment (miina) [reply assets/src/edit-story/components/form/filterToggle/stories/index.js]: 😕 Odd! Thanks for noting.

**Pull request merged**